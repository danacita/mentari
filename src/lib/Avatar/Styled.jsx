import styled, { css } from 'styled-components';

import Color from '../Styles/bases/Color';

const AvatarWrapper = styled.div`
  width: 40px;
  height: 40px;
  display: inline-flex;
  vertical-align: middle;
  background-color: ${Color.grey191};
  justify-content: center;
  align-items: center;
  ${props =>
    props.square
      ? css`
          border-radius: 6px;
        `
      : css`
          border-radius: 100px;
        `}
`;

const AvatarInitial = styled.span`
  font-weight: 600;
  color: ${Color.white};
`;

export { AvatarWrapper, AvatarInitial };
