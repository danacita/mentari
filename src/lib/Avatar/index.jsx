import React from 'react';
import PropTypes from 'prop-types';

import { AvatarWrapper, AvatarInitial } from './Styled';

const propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  profile: PropTypes.object.isRequired
};

const getInitials = string => {
  const names = string.split(' ');
  const initials = names[0].substring(0, 1).toUpperCase();
  return initials;
};

const Avatar = ({ profile }) => (
  <AvatarWrapper>
    <AvatarInitial>{getInitials(`${profile.firstName}`)}</AvatarInitial>
    <AvatarInitial>{getInitials(`${profile.lastName}`)}</AvatarInitial>
  </AvatarWrapper>
);

Avatar.propTypes = propTypes;

export default Avatar;
