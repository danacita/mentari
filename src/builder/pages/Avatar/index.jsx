import React, { Component, Fragment } from 'react';
import SyntaxHighlighter from 'react-syntax-highlighter';
import { atomOneDarkReasonable } from 'react-syntax-highlighter/dist/esm/styles/hljs';
import { ContentWithRightBar } from '../Styled';
import RightbarShared from '../../shared/Rightbar';
import DemoShared from '../../shared/Demo';
import PropsShared from '../../shared/Props';
import CodeShared from '../../shared/Code';
import { Margin, Avatar, Radio } from '../../../lib';

class AvatarPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      avatarShape: 'circle'
    };
  }

  render() {
    const { avatarShape } = this.state;
    const codeString = `import { Avatar } from '@danacita/mentari';\n<Avatar profile={{ firstName: 'Arnold', lastName: 'Sebastian' }} ${
      avatarShape === 'square' ? 'square ' : ''
    }/>`;
    return (
      <Fragment>
        <ContentWithRightBar>
          <h1>Avatar</h1>
          <p>Lorem ipsum dolor, sit amet consectetur adipisicing elit.</p>
          <Margin top={24}>
            <PropsShared>
              <li>
                <span>Profile</span>
                <span>Apply firstname and lastname with object</span>
              </li>
              <li>
                <span>Square</span>
                <span>Set the shape of Avatar</span>
              </li>
            </PropsShared>
          </Margin>
          <DemoShared>
            <h4>Demo</h4>
            <p>You can change the props via the component builder on your right</p>
            <Margin top={16}>
              <Avatar
                profile={{ firstName: 'Arnold', lastName: 'Sebastian' }}
                square={avatarShape === 'square'}
              />
            </Margin>
          </DemoShared>
          <h4>Code</h4>
          <Margin top={16}>
            <CodeShared copyCode={codeString}>
              <SyntaxHighlighter language="jsx" style={atomOneDarkReasonable} showLineNumbers>
                {codeString}
              </SyntaxHighlighter>
            </CodeShared>
          </Margin>
          <RightbarShared title="Avatar">
            <Margin bottom={16}>
              <h4>Shape of Avatar</h4>
            </Margin>
            <Margin bottom={24}>
              <Radio
                name="gender"
                fieldData={[{ id: 'circle', name: 'Circle' }, { id: 'square', name: 'Square' }]}
                withRedux={false}
                block
                defaultValue={avatarShape}
                onChange={e => this.setState({ avatarShape: e.target.id })}
              />
            </Margin>
          </RightbarShared>
        </ContentWithRightBar>
      </Fragment>
    );
  }
}

export default AvatarPage;
