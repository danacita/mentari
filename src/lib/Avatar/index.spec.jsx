import React from 'react';
import renderer from 'react-test-renderer';

import Avatar from '.';

describe('Avatar', () => {
  it('renders correctly', () => {
    const profile = {
      firstName: 'John',
      lastName: 'Doe'
    };
    const tree = renderer.create(<Avatar profile={profile} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
