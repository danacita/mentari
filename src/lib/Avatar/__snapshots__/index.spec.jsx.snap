// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<div
  className="sc-bdVaJa cijffQ"
  style={null}
>
  <span
    className="sc-bwzfXH cREbSS"
  >
    J
  </span>
  <span
    className="sc-bwzfXH cREbSS"
  >
    D
  </span>
</div>
`;
